packages/odie-client/src/hooks/use-get-combined-chat.ts

```diff
@@ -12,53 +12,59 @@ import type { Chat, Message } from '../types';
  * @returns The combined chat.
  */
 export const useGetCombinedChat = ( canConnectToZendesk: boolean ) => {
-	const { currentSupportInteraction, isChatLoaded } = useSelect( ( select ) => {
-		const store = select( HELP_CENTER_STORE ) as HelpCenterSelect;
-		return {
-			currentSupportInteraction: store.getCurrentSupportInteraction(),
-			isChatLoaded: store.getIsChatLoaded(),
-		};
-	}, [] );
+	const { currentSupportInteraction, conversationId, odieId, isChatLoaded } = useSelect(
+		( select ) => {
+			const store = select( HELP_CENTER_STORE ) as HelpCenterSelect;
+			const currentSupportInteraction = store.getCurrentSupportInteraction();
+
+			// Get the current odie chat ID
+			const odieId =
+				currentSupportInteraction?.events.find( ( event ) => event.event_source === 'odie' )
+					?.event_external_id ?? null;
+
+			// Get the current Zendesk conversation ID
+			const conversationId =
+				currentSupportInteraction?.events.find( ( event ) => event.event_source === 'zendesk' )
+					?.event_external_id ?? null;
+
+			return {
+				currentSupportInteraction,
+				conversationId,
+				odieId,
+				isChatLoaded: store.getIsChatLoaded(),
+			};
+		},
+		[]
+	);
 
 	const [ mainChatState, setMainChatState ] = useState< Chat >( emptyChat );
 	const getZendeskConversation = useGetZendeskConversation();
-	// Get the current odie chat
-	const odieId =
-		currentSupportInteraction?.events.find( ( event ) => event.event_source === 'odie' )
-			?.event_external_id ?? null;
-
-	// Get the current Zendesk conversation ID
-	const conversationId =
-		currentSupportInteraction?.events.find( ( event ) => event.event_source === 'zendesk' )
-			?.event_external_id ?? null;
 
 	const { data: odieChat, isLoading: isOdieChatLoading } = useOdieChat( Number( odieId ) );
 
 	useEffect( () => {
-		if ( odieId && ! conversationId ) {
-			if ( odieChat ) {
-				setMainChatState( {
-					...odieChat,
-					provider: 'odie',
-					conversationId: null,
-					supportInteractionId: currentSupportInteraction!.uuid,
-					status: 'loaded',
-				} );
-			}
-		} else if ( odieId && conversationId && canConnectToZendesk ) {
-			if ( odieChat && isChatLoaded ) {
+		if ( odieId && odieChat && ! conversationId ) {
+			setMainChatState( {
+				...odieChat,
+				provider: 'odie',
+				conversationId: null,
+				supportInteractionId: currentSupportInteraction!.uuid,
```

Comment:
We are seeing occasionally `currentSupportInteraction` is undefined, resulting a blank screen. I tried to dig it up but not too familiar with this part. Any ideas on top of your head? Thanks! @escapemanuele

Reply:
Yep so this crash the entire UI on my end as well.

Reply:
<img width="676" alt="image" src="https://github.com/user-attachments/assets/d186a430-f1b5-4ab6-9482-81607149fd8a" />

Reply:
CC - @agrullon95 @heavyweight

Reply:
Didn't see this, but this should be fixed!